package/MDAnalysis/topology/MOL2Parser.py

```diff
@@ -148,9 +150,34 @@ def parse(self, **kwargs):
         resids = []
         resnames = []
         charges = []
-
+        set_charge = True
+        for line in sections['molecule']:
+            if line.strip().lower() == 'no_charges':
+                set_charge = False
+                break
         for a in atom_lines:
-            aid, name, x, y, z, atom_type, resid, resname, charge = a.split()[:9]
+            columns = a.split()
+            if len(columns) >= 9:
+                columns = columns[:9]
+                aid, name, x, y, z, atom_type, resid, resname, charge = columns
+            elif len(columns) < 6:
+                raise ValueError(f"The @<TRIPOS>ATOM block in mol2 file"
+                                 f" {os.path.basename(self.filename)}"
+                                 f" should contain at least 6 fields to be"
+                                 f" unpacked: atom_id atom_name x y z"
+                                 f" atom_type [subst_id[subst_name"
+                                 f" [charge [status_bit]]]]")
+            else:
+                aid, name, x, y, z, atom_type = columns[:6]
+                resid = '1'
+                resname = ''
```

Comment:
**resname** is not essential for a Topology. Therefore, do not use a default (set it to `None`) and do *not* include it in the Topology.

See https://github.com/MDAnalysis/mdanalysis/issues/3385#issuecomment-1089067563

Reply:
@orbeckst Should I not include resnames attrubute if any of the resname is not provided?

Reply:
To be clear: *if* you have a resname then *do include resname in the topology*.

Reply:
Hi @orbeckst, thank you for the clarification. 

Considering the attributes are passed to `Topology` in a batch, would it be better to set `resname=None` for those atoms who didn't provide if other atoms in the file have a resname? Resnames attribute is not included only when all atoms are notprovided?

> To be clear: _if_ you have a resname then _do include resname in the topology_.

Reply:
I think our comments crossed path. I hadn't seen your question when I posted my last comment.

> Should I not include resnames attrubute if any of the resname is not provided?

The format definition is not clear about the situation where some lines might contain optional data and others don't. It _looks_ as if this was allowed for `@<TRIPOS>ATOM`:
```
atom_id atom_name x y z atom_type [subst_id [subst_name [charge [status_bit]]]]
```
However, that makes little sense for how MDAnalysis sees the data. Therefore, I propose we treat resname in the same way as charge: If even a single ATOM line is missing *resname* then we will not include *any* resname in the topology. I am also inclined to fail with `ValueError` if we find some lines with and others without resname. 

If someone with more experience with working with mol2 files wants to chime in then please do!

Reply:
I added a https://github.com/MDAnalysis/mdanalysis/issues/3385#issuecomment-1089582380 to the main issue.

Reply:
I will go with this one first.

> If even a single ATOM line is missing resname then we will not include any resname in the topology.

Reply:
> Considering the attributes are passed to Topology in a batch, would it be better to set resname=None for those atoms who didn't provide if other atoms in the file have a resname? Resnames attribute is not included only when all atoms are notprovided?

That seems to be a good idea: collect everything, then check if there are any `None` in the list. If all are `None` then do not include the TopoloAttr. If some are `None`, raise an exception an fail.